src/main/java/com/cst438/controller/AssignmentController.java

```diff
@@ -5,114 +5,280 @@
+            if(a == null) {
+                throw  new ResponseStatusException( HttpStatus.NOT_FOUND, "Assignment not found "+ dto.id());
+            } else {
+                Section section = sectionRepository.findById(dto.secNo()).orElse(null);
 
-        // TODO
-    }
+                if (section == null) {
+                    throw new ResponseStatusException(HttpStatus.NOT_FOUND, "section not found. secNo: "+dto.secNo());
+                } else if (section.getInstructorEmail() == null || !section.getInstructorEmail().equals(instructorEmail)) {
+                    throw new ResponseStatusException(HttpStatus.UNAUTHORIZED, "unauthorized");
+                }
 
-    // instructor gets grades for assignment ordered by student name
-    // user must be instructor for the section
-    @GetMapping("/assignments/{assignmentId}/grades")
-    public List<GradeDTO> getAssignmentGrades(@PathVariable("assignmentId") int assignmentId) {
+                Date startDate = new Date(section.getTerm().getStartDate().getTime());
+                Date endDate = new Date(section.getTerm().getEndDate().getTime());
+                Date dueDate;
 
-        // TODO remove the following line when done
+                try {
+                    SimpleDateFormat df =new SimpleDateFormat("yyyy-MM-dd");
+                    dueDate = df.parse(dto.dueDate());
+                } catch (ParseException e) {
+                    throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "assignment due date must be in yyyy-MM-dd format");
+                }
+                if (dueDate.before(startDate) || dueDate.after(endDate)) {
+                    throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "assignment due date must be within section term");
+                }
 
-        // get the list of enrollments for the section related to this assignment.
-		// hint: use te enrollment repository method findEnrollmentsBySectionOrderByStudentName.
-        // for each enrollment, get the grade related to the assignment and enrollment
-		//   hint: use the gradeRepository findByEnrollmentIdAndAssignmentId method.
-        //   if the grade does not exist, create a grade entity and set the score to NULL
-        //   and then save the new entity
+                a.setTitle(dto.title());
+                a.setDueDate(dto.dueDate());
+                assignmentRepository.save(a);
+                return new AssignmentDTO(
+                    a.getAssignmentId(),
+                    a.getTitle(),
+                    a.getDueDate(),
+                    a.getSection().getCourse().getCourseId(),
+                    a.getSection().getSecId(),
+                    a.getSection().getSectionNo()
+                );
+            }
+        }
 
-        return null;
-    }
+        // delete assignment for a section
+        // logged in user must be instructor of the section
+        @DeleteMapping("/assignments/{assignmentId}")
```

Comment:
do we need to add  @PreAuthorize in  deleteAssignment? I remember the professor mentioning all Controller methods need to be updated with @PreAuthorize

Reply:
Thanks @numbee12 . I pushed a commit that should resolve this issue. Let me know if it looks good.